modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1522,6 +1522,11 @@ CV__DNN_INLINE_NS_BEGIN
           *  @param[out] mask Allocated class prediction for each pixel
           */
          CV_WRAP void segment(InputArray frame, OutputArray mask);
+
+         /** @brief select which output layer to use as the output of the model
+          *  @param[in] outname the name of the output layer
+          */
+         CV_WRAP void selectOutput(const std::string &outname);
```

Comment:
@vpisarev @opencv-alalek @dku What do you think about the API change? Should be port it to 4.x?

Reply:
Should it be added to a base `Model` class? I think it might be useful for different types of models.

Reply:
I added this method is used to select one specific output in the model. The model will output extra data for the auxiliary. Refrence: https://github.com/onnx/models/tree/main/validated/vision/object_detection_segmentation/fcn

> The model has two outputs, ("out", "aux"). "out" is the main classifier and has shape (N, 21, height, width). Each output pixel is one-hot encoded, i.e. np.argmax(out[image, :, x, y]) is that pixel's predicted class. Class 0 is the background class